### ISSUE TITLE
A way to simulate background-size: contain

### ISSUE THREAD

**Issue by ErfanEbrahimnia** [konvajs/react-konva#568]
Is there a way to make image scaling behave like `background-size:`/ `object-fit:` `contain`? I have already seen the example with `cover` behaviour but couldn't find one for `contain` 🤔

**Comment by lavrton** [comment]
You just need to calculate the required scale for the image and manually place it inside the center of the specified width/height. Doesn't look hard to me.

**Comment by ErfanEbrahimnia** [comment]
Just to make sure that we mean the same `contain` mode because for what I want to achieve the Konva node of the image has to stay the same size for each input image but the image itself has to be scaled the same way it would be for example inside a 100x100 `div` with `background-size: cover`.
If that's possible with Konva could you please give me a hint on how to achieve such a behaviour

**Comment by lavrton** [comment]
> because for what I want to achieve the Konva node of the image has to stay the same size

I don't think it is possible with Konva API. You have to resize/scale the image node itself. 

> If that's possible with Konva could you please give me a hint on how to achieve such a behaviour

```js
const areaWidth = 100;
const areaHeight = 100;

const scaleX = areaWidht / image.width();
const scaleY = areaHeight / image.height();
const scale = Math.min(scaleX, scaleY);

image.setAttrs({
  x: (areaWidth - image.width() * scale) / 2,
  y: (areaHeight - image.height() * scale) / 2,
  scaleX: scale,
  scaleY: scale
});
```

**Comment by ErfanEbrahimnia** [comment]
> I don't think it is possible with Konva API
Shoot

Is there way to achieve this behavior with a container or group wrapping the image node?
I'm new to Konva so sorry if I am bothering you with basic questions. Also sorry for opening this issue inside the react repo

**Comment by lavrton** [comment]
> Is there way to achieve this behavior with a container or group wrapping the image node?

Sure. As I mentioned you just need to manually move/scale `Konva.Image` instance. Just need to recalculate scale/position for each input image. The code above does that.

**Comment by ErfanEbrahimnia** [comment]
Got it, I will try that. Just one final question. Do you think it's also possible to implement the `cover` behaviour with this method? Basically using the `Container` Node as the bounding box of the image opposed to the recommended method in the examples which uses `crop`?

**Comment by ErfanEbrahimnia** [comment]
Ok, thanks a lot for taking the time to explain ♥️
